metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchQueryBuilder.java

```diff
@@ -84,13 +87,15 @@ private QueryBuilder buildInternalQuery(@Nonnull List<EntitySpec> entitySpecs, @
     return finalQuery;
   }
 
-  private static Set<SearchFieldConfig> getStandardFields(@Nonnull EntitySpec entitySpec) {
+  private Set<SearchFieldConfig> getStandardFields(@Nonnull EntitySpec entitySpec) {
     Set<SearchFieldConfig> fields = new HashSet<>();
 
     // Always present
     final float urnBoost = Float.parseFloat((String) PRIMARY_URN_SEARCH_PROPERTIES.get("boostScore"));
-    List.of("urn", "urn.delimited").forEach(urnField -> fields.add(SearchFieldConfig.detectSubFieldType(
-            urnField, urnBoost, SearchableAnnotation.FieldType.URN)));
+
+    fields.add(SearchFieldConfig.detectSubFieldType("urn", urnBoost, SearchableAnnotation.FieldType.URN));
+    fields.add(SearchFieldConfig.detectSubFieldType("urn.delimited", urnBoost * partialConfiguration.getUrnFactor(),
```

Comment:
why is this important

Reply:
Will do a full detailed RCA on this, however the tldr is that a smoke test was dependent on a `container` entity result proceeding several `dataset` results, otherwise the expected container is on the next page. The `datasets` were included based on a partial match on the `urn` field and a `container` match was from a partial match on the `name` field (other fields were involved which makes it a bit more complicated). The urn match was pulling the datasets ahead of the container results.

In general, a partial match on non-Urn fields is reduced by 0.4, however an urn partial match was not being penalized in the same way. Both are now configurable and ultimately it was reasoned that a partial match on an urn should be slightly better than a partial match on a non-Urn field, thus 0.7 (urn partial) vs 0.4f (non-urn partial).

Note that because of the base scoring for hits is designed for needle in haystack problems it uses the typical statistical tf/idf function which is subject to differences in data.These configuration weights and values were recently adjusted for one particular set of data and better field to query matching increased the importance of partial matching whereas previously the code was focused on prefix/exact matching.